### PR TITLE
JSON Web Tokens

### PR DESCRIPTION
**Pull request by mmussomele**
Tried on the basic api_authz example. I modified echo_server to get the encoded token from the request, and changed the policy to only allow a request if a token's payload mapped the "iss" key to "opa". Tested by issuing the requests (the `?` may need to be escaped):
```
$ curl --user alice:password localhost:5000/finance/salary/alice?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIwMTIzNDUiLCJpc3MiOiJtYXR0In0.KRP5pIh4f5Ksn7L3_osDg6rTjbFL5zv7QUcH_EK3f4U
Error: user alice is not authorized to GET url /finance/salary/alice
$ curl --user alice:password localhost:5000/finance/salary/alice?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIwMTIzNDUiLCJpc3MiOiJvcGEifQ._VInZhCqslB-Q77uBd0sq37hJE66URQ-FBVKHVLm-Xo
Success: user alice is authorized
```
The first token has "iss" mapped to "matt", and the second has "iss" mapped to "opa".

### CODE REVIEW

File: topdown/tokens.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	o, ok := v.(ast.Object)
+	if !ok {
+		return nil, errors.New("decoded JSON type was not an Object")
+	}
+
+	return o, nil
+}
+
+func init() {
+	wrap := func(t *Topdown, expr *ast.Expr, iter Iterator) error {
```
Comment (tsandall) [inline topdown/tokens.go]: Let's move this into builtins.go and make it available along with the other functional built-in wrappers. I realize that single input, triple output built-ins are rare but this is useful nonetheless.

**Comment by mmussomele** [comment]
Updated

**Comment by mmussomele** [comment]
ah wait, forgot to change the names

**Comment by mmussomele** [comment]
ok all good now

**Pull request merged**